### PR TITLE
flavours: add module

### PR DESCRIPTION
**Pull request by Misterio77**
### Description

[flavours](https://github.com/misterio77/flavours) is a [base16](https://github.com/chriskempson/base16) manager. It allows you to instantly theme any supported configuration file (using [base16 templates](https://github.com/chriskempson/base16#template-repositories)) with more than 180 existing [base16 schemes](https://github.com/chriskempson/base16#scheme-repositories), or even generate these from images.

Please let me know if there's anything missing, i'm still learning about Home Manager and Nix :D

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/programs/flavours.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+  cfg = config.programs.flavours;
+
+  tomlFormat = pkgs.formats.toml { };
+
+in {
+  meta.maintainers = [ maintainers.misterio77 ];
+
+  options.programs.flavours = {
+    enable = mkEnableOption "Flavours";
```
Comment (sumnerevans) [inline modules/programs/flavours.nix:15]:
```suggestion
    enable = mkEnableOption "Flavours, a base16 manager";
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]
Looks great overall. Just a small comment. Also, this needs an entry in `modules/misc/news.nix`.

File: modules/programs/flavours.nix (hunk 1 above)
Comment (Misterio77) [reply modules/programs/flavours.nix]: Thank you!

File: modules/programs/flavours.nix [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+              subtemplate = "colors";
+              hook = "swaymsg reload";
+              rewrite = true;
+            }
+          ];
+        }
+      '';
+      description = ''
+        Configuration written to
+        <filename>~/.config/flavours/config.toml</filename>.
```
Comment (sumnerevans) [inline modules/programs/flavours.nix:50]:
```suggestion
        <filename>$XDG_CONFIG_HOME/flavours/config.toml</filename>.
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]
sorry, one more thing

File: modules/programs/flavours.nix (hunk 2 above)
Comment (Misterio77) [reply modules/programs/flavours.nix]: No problem, thanks for the fix!

**Review by sumnerevans — APPROVED** [review]

**Comment by berbiche** [comment]
Would it be possible/make sense to generate the configuration during the build as opposed to the runtime?

The generated configuration would then be part of the config and more reproducible.

If so, it could be enabled with an option.

**Comment by Misterio77** [comment]
I might be wrong, but don't all home-manager modules work that way? I mean, the derivations are evaluated (thus, the configs are generated) when the system is built? I'm doing it the standard way, i think

**Comment by Misterio77** [comment]
> Would it be possible/make sense to generate the configuration during the build as opposed to the runtime?

Oh i get it now. You mean the applications flavours is theming, not flavours itself, right?

Yeah, flavours is a little stateful, i think. It does work nicely when you theme a mutable config and import it in your immutable one, for example.

I'm actually porting my config to use some of the ideias from https://github.com/nix-community/home-manager/issues/361 right now, but still needs some work before it gets as seamless as flavours. The main one being getting `onChange` working on more applications.

I'm hacking at these right now:
- alacritty (usually live reloads, but home-manager swaps symlinks, which [don't get watched](https://github.com/alacritty/alacritty/issues/4590), but [got it working](https://github.com/Misterio77/home-manager/commit/d531704e8a25afc8c7d5db1848424b2dfaf03646) with [this patch](https://github.com/ncfavier/alacritty/commit/5f392c2cb516a5ea198ebb48754c7c42157d21b3))
- [gtk](https://github.com/nix-community/home-manager/pull/2169)
- qutebrowser (just merged wohoo)
- waybar

I think i might close this PR and focus on this approach, instead.

**Comment by berbiche** [comment]
Sorry, I forgot to reply.

There are multiple modules (neovim, xmonad, etc.) generating additional configuration during the build and linking files to the appropriate place (e.g. setting `home.file`).

What I was asking was whether Flavors could generate its configuration files for the different programs while evaluating the HM configuration.

The advantage of doing so would be reproducibility (as long as Flavors is reproducible) and caching of the built configuration (the derivation building the config files).

**Comment by Misterio77** [comment]
No problems!

Everything flavours actually does is writing stuff to files in-place, so i think it won't play nicely with the setups most people using home-manager usually have. Replacing colors at build time with home-manager actually makes flavours obsolete haha.

**Comment by berbiche** [comment]
Side note, Alacritty supports imports: https://github.com/alacritty/alacritty/blob/72675db3826ced63990fdede3dcaf07e76b1dde7/alacritty.yml#L3-L12

**Pull request closed without merging**